src/test/resources/processor_test_files/S1068_UnusedPrivateField/UnusedPrivateField.java

```diff
@@ -0,0 +1,12 @@
+import java.io.Serializable;
+
+public class UnusedPrivateField implements Serializable {
+    private static final long serialVersionUID = 42L; // Compliant as this field is used in deserialization
```

Comment:
This field was not deleted even though I **did not** implement the class using `Serializable` initially. I think this could be a bug in the sonar scanner for violations. What do you think, @monperrus @slarse ?

Reply:
You could have a look inside of the `UnusedPrivateFieldCheck` and see how they select fields. Possibly, they just ignore any field with that name.

Reply:
[It](https://github.com/SonarSource/sonar-java/blob/a5947abe43d79a638c61413613b21f417fa9f400/java-checks/src/main/java/org/sonar/java/checks/unused/UnusedPrivateFieldCheck.java#L144-L147) checks for modifiers apart from checking for name. I could propose if they want check for class implementation too but they can probably dismiss this proposal as trivial because `serialVersionUID` is a semantically reserved word in Java.